Classes/ContentObject/Relation.php

```diff
@@ -183,27 +183,37 @@ protected function getRelatedItemsFromMMTable($localTableName, $localRecordUid,
 
         $relatedRecords = $this->getRelatedRecords($foreignTableName, ...$selectUids);
         foreach ($relatedRecords as $record) {
-            if (isset($foreignTableTca['columns'][$foreignTableLabelField]['config']['foreign_table'])
-                && $this->configuration['enableRecursiveValueResolution']
-            ) {
-                if (strpos($this->configuration['foreignLabelField'], '.') !== false) {
-                    $foreignTableLabelFieldArr = explode('.', $this->configuration['foreignLabelField']);
-                    unset($foreignTableLabelFieldArr[0]);
-                    $this->configuration['foreignLabelField'] = implode('.', $foreignTableLabelFieldArr);
-                }
-
-                $this->configuration['localField'] = $foreignTableLabelField;
-
-                $contentObject = GeneralUtility::makeInstance(ContentObjectRenderer::class);
-                $contentObject->start($record, $foreignTableName);
-
-                return $this->getRelatedItems($contentObject);
-            } else {
-                if (Util::getLanguageUid() > 0) {
-                    $record = $this->frontendOverlayService->getOverlay($foreignTableName, $record);
+            $itemParts = [];
+            foreach (explode(',', $foreignTableLabelField) as $fieldName) {
+                $fieldName = trim($fieldName);
+
+                if (isset($foreignTableTca['columns'][$fieldName]['config']['foreign_table'])
+                    && $this->configuration['enableRecursiveValueResolution']
+                ) {
+                    if (strpos($this->configuration['foreignLabelField'], '.') !== false) {
+                        $foreignTableLabelFieldArr = explode('.', $this->configuration['foreignLabelField']);
+                        unset($foreignTableLabelFieldArr[0]);
+                        $this->configuration['foreignLabelField'] = implode('.', $foreignTableLabelFieldArr);
+                    }
+
+                    $this->configuration['localField'] = $fieldName;
+
+                    $contentObject = GeneralUtility::makeInstance(ContentObjectRenderer::class);
+                    $contentObject->start($record, $foreignTableName);
+
+                    $itemParts = array_merge($itemParts, $this->getRelatedItems($contentObject));
+                } else {
+                    if (Util::getLanguageUid() > 0) {
+                        $record = $this->frontendOverlayService->getOverlay($foreignTableName, $record);
+                    }
+                    $itemParts[] = $record[$fieldName];
                 }
-                $relatedItems[] = $record[$foreignTableLabelField];
             }
+            $glue = ' ';
+            if (array_key_exists('foreignLabelFieldGlue', $this->configuration)) {
```

Comment:
Just a proposal: Why not use the null coalescing operator here(https://www.php.net/manual/de/migration70.new-features.php)? Could be then done in one line.